kafka-connect-s3/src/main/java/io/confluent/connect/s3/TopicPartitionWriter.java

```diff
@@ -208,7 +213,9 @@ public void write() {
               setNextScheduledRotation();
               nextState();
             } else {
-              currentEncodedPartition = encodedPartition;
+              if (!appendLateData || currentEncodedPartition == null) {
```

Comment:
Tbh, a comment wouldn't probably hurt here. Especially, since we weren't checking for `null` before